app/src/lib/stores/helpers/infer-comparison-branch.ts

```diff
@@ -0,0 +1,129 @@
+import { Branch, IAheadBehind } from '../../../models/branch'
+import { PullRequest } from '../../../models/pull-request'
+import { getAheadBehind, revRange } from '../../git'
+import { Repository } from '../../../models/repository'
+import { GitHubRepository } from '../../../models/github-repository'
+
+/**
+ * Infers which branch to use as the comparison branch
```

Comment:
That's something I thought about, but decided against since the inner workings of this function should be encapsulated. My thinking is we shouldn't leak implementation details as that would require the caller to understand how the branch is determined. I was deliberate in including the word `infer` to communicate that. 

Now, I'm not against adding how this is calculated to the function's JSdoc, but I would be interested in knowing any scenarios where having knowledge of this process would be beneficial.